src/helpers/misc.ts

```diff
@@ -158,3 +158,60 @@ export function runOnceAndCache<T, Args extends any[]>(
     return result ?? (result = func(...args));
   };
 }
+
+export function deriveCompactSignature(signature: string): string {
+  // Remove "0x" prefix if present
+  if (signature.startsWith('0x')) {
+    signature = signature.slice(2);
+  }
+
+  // Convert the hex string to a byte array
+  const bytes = new Uint8Array(signature.length / 2);
+  for (let i = 0; i < signature.length; i += 2) {
+    bytes[i / 2] = parseInt(signature.substr(i, 2), 16);
+  }
+
+  // Validate the signature length (64 or 65 bytes)
+  if (bytes.length !== 64 && bytes.length !== 65) {
+    throw new Error('Invalid signature length: must be 64 or 65 bytes');
+  }
+
+  // Extract r and s components
+  const r = `0x${Array.from(bytes.slice(0, 32), (b) =>
+    b.toString(16).padStart(2, '0')
+  ).join('')}`;
+  let v;
+
+  // Handle 64-byte (EIP-2098 compact) and 65-byte signatures
+  if (bytes.length === 64) {
+    // Extract v from the highest bit of s and clear the bit in s
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    v = 27 + (bytes[32]! >> 7);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    bytes[32]! &= 0x7f; // Clear the highest bit
+  } else {
+    // Extract v directly for 65-byte signature
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    v = bytes[64]!;
```

Comment:
Is it expected that I don't have linter errors here without disabling?

Reply:
Strangely, I also don't have errors anymore :shrug: